test/tests/functional/pbs_cray_vnode_pool.py

```diff
@@ -123,27 +123,21 @@ def test_two_moms_single_vnode_pool(self):
         self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA, attrib=attr)
         self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostB, attrib=attr)
 
-        rc = self.server.log_match("Mom %s added to vnode_pool %s" %
-                                   (self.hostB, '1'), max_attempts=5,
-                                   starttime=start_time)
-        self.assertTrue(rc)
+        self.server.log_match("Mom %s added to vnode_pool %s" %
+                              (self.hostB, '1'), max_attempts=5,
+                              starttime=start_time)
 
-        rc = self.momA.log_match(
-            "Hello (no inventory required) from server", max_attempts=9,
-            starttime=start_time)
-        if rc is None:
-            found_in_momA = 0
-        else:
+        _msg = "Hello (no inventory required) from server"
+        try:
+            self.momA.log_match(_msg, max_attempts=9, starttime=start_time)
             found_in_momA = 1
-
-        rc = self.momB.log_match(
-            "Hello (no inventory required) from server", max_attempts=9,
-            starttime=start_time)
-        if rc is None:
-            found_in_momB = 0
-        else:
+        except PtlLogMatchError:
+            pass
```

Comment:
I don't think, because at tests beginning found_in_momA and found_in_momB is initialized with 0.
If log_match doesn't match, then it will throw PTLlogMatchError and found_in_momA will not be set to 1.

Reply:
You are correct.  Raising the exception will stop execution and it'll never reach the assignment.  I'd rather see the variables set in the except portion.  This makes it very clear what you are doing.